core/src/main/java/org/keycloak/crypto/integration/DefaultCryptoProvider.java

```diff
@@ -0,0 +1,23 @@
+package org.keycloak.crypto.integration;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import org.keycloak.jose.jwe.alg.AesKeyWrapAlgorithmProvider;
+import org.keycloak.jose.jwe.alg.JWEAlgorithmProvider;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class DefaultCryptoProvider implements CryptoProvider {
```

Comment:
Is the plan to move this to crypto/default later? Or should it remain here permanently?

Reply:
Yes, the plan is to do it later. Moving it now will mean that module "crypto/default" will be new mandatory module for the default distribution (including adapters), which will mean lots of additional changes in this PR in the packaging etc.

IMO The ideal final state will be, that modules like keycloak-common or keycloak-core don't have dependency on bouncyCastle and hence the crypto stuff is incrementally moved into "crypto/default" and "crypto/fips" and at this stage the DefaultCryptoProvider will be in "crypto/default" .